src/pybind/mgr/dashboard/services/ceph_service.py

```diff
@@ -120,12 +120,24 @@ def get_pool_list_with_stats(cls, application=None):
 
     @classmethod
     def get_pool_name_from_id(cls, pool_id):
+        pool = cls.get_pool_by_attribute('pool', pool_id)
+        return pool['pool_name'] if pool is not None else None
+
+    @classmethod
+    def get_pool_by_attribute(cls, attribute, value):
         pool_list = cls.get_pool_list()
         for pool in pool_list:
-            if pool['pool'] == pool_id:
-                return pool['pool_name']
+            if pool[attribute] == value:
+                return pool
         return None
 
+    @classmethod
+    def get_pool_pg_status(cls, pool_name):
+        pool = cls.get_pool_by_attribute('pool_name', pool_name)
+        if pool is None:
+            return None
```

Comment:
```suggestion
            return []
```
otherwise you'll get a `TypeError: argument of type 'NoneType' is not iterable?` in `if 'creating+incomplete' in pg_status:`.

Reply:
Done returning an empty dict.